### ISSUE TITLE
[FE] 10-10EZ Short-form: Add labels for Edit Dependent button(s) > [Markup and meta data] Visual labels aren't included in the accessible label. (09.05.3)

### ISSUE THREAD

**Issue by hdjustice** [department-of-veterans-affairs/va.gov-team#42856]
@shiragoodman commented on [Tue Apr 26 2022](https://github.com/department-of-veterans-affairs/va-forms-system-core/issues/358)

#### VFS team name
10-10 Team

#### VFS product name
Health Care Application 10-10EZ

#### VFS feature name
10-10EZ Short Form (Auto-qualifiers)

#### Point of Contact/Reviewers
Brian DeConinck (@briandeconinck) - Accessibility

*For more information on how to interpret this ticket, please refer to the [Anatomy of a Staging Review issue ticket](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/Anatomy-of-a-Staging-Review-Issue-ticket.2060320997.html) guidance on Platform Website.

---

### Platform Issue
Visual labels aren't included in the accessible label.

### Issue Details
On Step 4 of 6 [disability rating <50%]: Household Information, when a user is adding multiple dependents, an Edit button is paired with each dependent. Each of these Edit buttons have identical accessible names, "Edit Dependent." For screen reader users tabbing through the buttons, it's difficult to identify which dependent will be edited by following which button.        

### Link, screenshot or steps to recreate
1. Log into the RI as user 11.
2. Complete the form up to Step 4 of 6.
3. For "Do you have any dependents to report?" answer Yes.
4. Enter information for a dependent.
5. Click "Add another Dependent"
6. Enter information for a dependent.
7. Click ""Add another insurance policy.""
8. Activate a screen reader (recommend NVDA on Windows or VoiceOver on Mac for testing).
9. Press [tab] or [shift] + [tab] to move forward and backward through the interactive elements on the page.
10. Note that the edit buttons for each depdendent are announced only as "Edit Dependent."

### VA.gov Experience Standard
[Category Number 09, Issue Number 05](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/VA.gov-experience-standards.1683980311.html)

### Other References
WCAG SC 2.5.3_A

---

### Platform Recommendation
Each edit button should have a unique accessible name. This can be accomplished either by changing the button text (`<button>Edit [name of dependent]</button>`) or by providing a unique aria-label (`<button aria-label="Edit [name of dependent]">Edit</button>`).

### VFS Team Tasks to Complete

- [ ] Comment on the ticket if there are questions or concerns
- [ ] Close the ticket when the issue has been resolved or validated by your Product Owner. If a team has additional questions or needs Platform help validating the issue, please comment in the ticket. 

---

@hdjustice commented on [Thu Apr 28 2022](https://github.com/department-of-veterans-affairs/va-forms-system-core/issues/358#issuecomment-1143750648)

@briandeconinck 
The 10-10EZ short form skips over this question, and does not appear during this process.  We agree that this does need to be addressed sooner rather than later, however this is in production now and we feel it should not be a launch blocker to functionality that does not display it.

---

@lzapson commented on [Wed May 04 2022](https://github.com/department-of-veterans-affairs/va-forms-system-core/issues/358#issuecomment-1143752595)

This is a forms issues, we cannot change the aria label in the application code when you can add multiple items of the same kind, like dependents. You can find the platform code that is rendering the edit button on the Household Information page, [here](https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/platform/forms-system/src/js/fields/ArrayField.jsx#L339).

This is also happening on the Review Application page, when a user edits Dependent information, which is also a forms issue. You can find the platform code that is rendering the Dependent information edit button on the Review Application page, [here](https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/platform/forms-system/src/js/review/ObjectField.jsx#L170).

@shiragoodman @hdjustice  please assign to the appropriate team. 

---

@hdjustice commented on [Tue May 24 2022](https://github.com/department-of-veterans-affairs/va-forms-system-core/issues/358#issuecomment-1143752599)

>This is a forms issues, we cannot change the aria label in the application code when you can add multiple items of the same kind, like dependents. You can find the platform code that is rendering the edit button on the Household Information page, here.
>
>This is also happening on the Review Application page, when a user edits Dependent information, which is also a forms issue. You can find the platform code that is rendering the Dependent information edit button on the Review Application page, here.
>
>@shiragoodman @hdjustice please assign to the appropriate team.

@shiragoodman - Hey there!  This issue is the same as in department-of-veterans-affairs/va-forms-system-core#357 that I just commented on.  How should we handle this ticket, since it is a forms component issue, and not specific to our coding?

---

@shiragoodman commented on [Tue May 24 2022](https://github.com/department-of-veterans-affairs/va-forms-system-core/issues/358#issuecomment-1143752612)

hey @hdjustice. I was just following up on this and 40490 yesterday (I had asked Lisa if another ticket was similar). Typically when this occurs, we remove the teams gh label and add the responsible team. We applied `vsa-10-10` as your team label when this ticket was created, but I noticed the `1010-team` label was also added. Which team label are you using?

---

@hdjustice commented on [Tue May 24 2022](https://github.com/department-of-veterans-affairs/va-forms-system-core/issues/358#issuecomment-1143752616)

Hi @shiragoodman - Since we are no longer on the VSA contract, and to avoid changing it in the future, we created a more ambiguous label of `1010-team`
I am glad you brought this up, as I was thinking of our conversation with the My VA team labels, and wanted to confirm with you on the product or team labels that the governance team would be using.

---

@shiragoodman commented on [Tue May 24 2022](https://github.com/department-of-veterans-affairs/va-forms-system-core/issues/358#issuecomment-1143753700)

@hdjustice we use the labels that you tell us to use. We don't want to create new labels for your team/product if labels already exist. For this ticket, we applied `vsa-10-10` as the team label and `vsa-hca` as the product label. 

If you don't like the labels that are currently in use, I'd recommend changing the existing labels instead of creating new ones. That way, all existing tickets with those labels get updated. 

So would you like us to use `1010-team` as your teams preferred team label moving forward?

---

@hdjustice commented on [Tue May 24 2022](https://github.com/department-of-veterans-affairs/va-forms-system-core/issues/358#issuecomment-1143753706)

@shiragoodman - You are right.  The  `vsa-10-10 ` label should have been renamed to **`1010-team`**.  This is our new team label and will always be on our tickets.
Since we have 2 products, we have a label for each = **`1010-ez`** and **`1010-cg`** 
We plan to use these on tickets specific to each one.   We may have other labels specific to the feature, but we don't want to muddy up things for the governance team.

I will work on getting the labels updated properly and remove the duplicates

---

@shiragoodman commented on [Tue May 24 2022](https://github.com/department-of-veterans-affairs/va-forms-system-core/issues/358#issuecomment-1143753712)

sounds good, @hdjustice ! thanks for the updates. I will get the team label updated.

---

@shiragoodman commented on [Tue May 24 2022](https://github.com/department-of-veterans-affairs/va-forms-system-core/issues/358#issuecomment-1143753718)

@hdjustice removed your team label.

@caw310 @k80bowman reassigning to Design System team. 

---

@taylorkaren commented on [Fri May 27 2022](https://github.com/department-of-veterans-affairs/va-forms-system-core/issues/358#issuecomment-1143753722)

From the 5/27/22 Design System Team/Forms Library Team sync call: Keep this ticket in mind while DST is working on the button component.

---

@taylorkaren commented on [Wed Jun 01 2022](https://github.com/department-of-veterans-affairs/va-forms-system-core/issues/358#issuecomment-1143757726)

Transferred to https://github.com/department-of-veterans-affairs/va-forms-system-core/issues/358.

**Comment by hdjustice** [comment]
Moved ticket to 10-10 Health Apps space to address and close out.

**Comment by hdjustice** [comment]
Per the comment left on issue 357 regarding this ticket (converted from ticket 358, and ticket #40490 (same functionality):

> **Micahchiang** 
> Update: @obliviga has a pull request out that will fix this specifically for the 10-10EZ. We discovered the functionality for unique aria-labels already exists, but wasn't documented anywhere. We'll be updating our documentation to include an entry for itemAriaLabel, which accepts a function that takes formdata as it's parameter and can return the name for the individual entry.
> 
> Edit: Since this functionality is readily available, #358 should be addressable by whichever application team is in charge of the 10-10ez.

**Comment by hdjustice** [comment]
@briandeconinck - This ticket moved around a bit since we needed some help from the design system team.  It has come back to us to address, however I still have the same comment:

> The 10-10EZ short form skips over this question, and does not appear during this process. We agree that this does need to be addressed sooner rather than later, however this is in production now and we feel it should not be a launch blocker to functionality that does not display it.
>Can you give us your thoughts on whether this has to be a launch blocker to the Short Form release? Or can we address it separately as a fast follow?

cc: @markfallows

**Comment by taylorkaren** [comment]
@hdjustice : Thanks for picking up on the comment on our ticket 357 and getting the work from our ticket back on the 10-10 team track. I've closed our 358.

**Comment by obliviga** [comment]
Just wanted to add some supplemental information here. Once [21312](https://github.com/department-of-veterans-affairs/vets-website/pull/21312/) is merged, the appropriate team can use the following code, if they wish, to address the issue outlined in this ticket: 

<img width="538" alt="image" src="https://user-images.githubusercontent.com/5934582/173685883-1e184c5a-7edf-42d1-92ce-3d2d910252ae.png">

cc @taylorkaren @hdjustice

**Comment by hdjustice** [comment]
Thank you @obliviga and @taylorkaren 
We have the same issue with #40490, but it is for Insurance information edit buttons, in the case where someone has more than one insurance policy.  I am assuming we could use a similar pattern, replacing the Name (first or last) with the "Name of the Insurance Provider".

**Comment by shiragoodman** [comment]
@hdjustice  - following up on this per [your comments](https://github.com/department-of-veterans-affairs/va.gov-team/issues/38031#issuecomment-1372292985) on #38031. I am not sure I followed the comment trail here... is this an issue 10-10 team is responsible for or is a Design System concern?

**Comment by hdjustice** [comment]
@shiragoodman We worked with the Forms team to determine the best way to address the issue of editing a dependent item (or insurance item in the other ticket).  Once they were able to identify a functionality for itemAriaLabel and help us understand how it works, the tickets came back to us to implement.  
I think the main take-away was that this "itemAriaLabel" function was not documented anywhere.  It looks like Micah may have added it to some documentation, but I could not say where.

**Comment by shiragoodman** [comment]
@hdjustice so it sounds like it's still an open issue. When does your team plan to implement?

**Comment by hdjustice** [comment]
@shiragoodman Great News!  After confirming with the team, this update has already been implemented for both the Dependents (this ticket) and Insurance sections (https://github.com/department-of-veterans-affairs/va.gov-team/issues/38031) of the form!  Screenreaders will now announce the name of the Dependent and Insurance companies when tabbing through the Edit buttons

**Comment by obliviga** [comment]
@hdjustice `itemAriaLabel` was documented here by me: https://depo-platform-documentation.scrollhelp.site/developer-docs/va-forms-library-about-schema-and-uischema

If you do a search for  `itemAriaLabel` on that page, you'll find it. Thank you for looking into this!

**Comment by shiragoodman** [comment]
that's fantastic @hdjustice ! Thank you for looking into it.

**Comment by hdjustice** [comment]
Thank you @obliviga!
